File: Java/AlgoExpert/Recursion/Medium/Staircase Traversal.java
import java.util.*;

class Program {

  public int staircaseTraversal(int height, int maxSteps) {
    // Write your code here.
    if (height == 0) return 1;
    int numberOfWays = 0;
    
    for (int i = 1; i <= maxSteps; i++) {
      if (i > height) break;
      int childCount = staircaseTraversal(height - i, maxSteps);
      numberOfWays += childCount;
    }
    
    return numberOfWays;
  }

  
}

File: Python/AlgoExpert/Recursion/Medium/staircaseTraversal.py
def staircaseTraversal(height, maxSteps):
    # Write your code here.
    if height == 0:
        return 1
    elif height < 0:
        return 0

    ways = 0
    for i in range(1, maxSteps + 1):
        ways += staircaseTraversal(height - i, maxSteps)
    
    return ways
